sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/DiagnosticsProvider.java

```diff
@@ -486,7 +481,11 @@ private void handleDiagnostics(Context context, CosmosDiagnosticsContext cosmosC
         // but there is some risk given that diagnostic handlers are custom code of course
         if (this.diagnosticHandlers != null && this.diagnosticHandlers.size() > 0) {
             for (CosmosDiagnosticsHandler handler: this.diagnosticHandlers) {
-                handler.handleDiagnostics(cosmosCtx, context);
+                try {
+                    handler.handleDiagnostics(cosmosCtx, context);
+                } catch (Exception e) {
+                    LOGGER.warn("HandledDiagnostics failed. ", e);
```

Comment:
I would argue this should be logged as error

Reply:
updated